src/tests/ftest/control/dmg_system_leader_query.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2021 Intel Corporation.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. 8F-30005.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+from __future__ import print_function
+
+from control_test_base import ControlTestBase
+
+class DmgSystemLeaderQueryTest(ControlTestBase):
+    # pylint: disable=too-many-ancestors
+    """Test Class Description:
+    Verify the system leader-query function of the dmg tool.
+    :avocado: recursive
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(DmgSystemLeaderQueryTest, self).__init__(*args, **kwargs)
+        self.hosts = None
+
+    def setUp(self):
+        super(DmgSystemLeaderQueryTest, self).setUp()
+        self.hosts = self.dmg.hostlist
+
+    def test_dmg_system_leader_query(self):
+        """
+        JIRA ID: DAOS-4822
+        Test Description: Test that system leader-query command reports leader
+            consistently regardless of which server is queried.
+        :avocado: tags=all,small,daily_regression,hw,dmg,system_leader_query
```

Comment:
Sorry I didn't notice this during the first review, but if you use small,hw, it'll run on a 2-host system, but you specified 3 servers in yaml, so the test will not work. Also, it doesn't look like you need hardware nodes for this test, so you can remove small and hw from the tags list and the test will run on a VM system.

Also, if you want to run only this test, you should skip all the hardware tests.
Skip-func-hw-test: true

You can also skip the unit tests to speed things up.
Skip-unit-test: true

Reply:
Good catch, thanks. I didn't realize the small tests were only 2 hosts. I'll fix the tags and repush.